woocommerce/woocommerce
# Fix Stock status behavior

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR modifies the Stock Indicator block to ensure it matches the product stock status.

Currently, some product stock statuses don't match between the `Product Collection` front end and the product page.


Errors:

**Simple Product**
| Status | Collection Frontend | Product page |
|--------|--------|--------|
| In stock / No stock management| ![Screenshot 2024-12-05 at 5 11 40 PM](https://github.com/user-attachments/assets/6734d4bf-c59a-438a-9793-1e2552c48f7f) | ![Screenshot 2024-12-05 at 5 12 08 PM](https://github.com/user-attachments/assets/c3d1a786-af97-44f0-83ef-07b985e6be0c) |
| Quantity 0 / Allow backorders (no notify) / Stock management | ![Screenshot 2024-12-05 at 5 15 45 PM](https://github.com/user-attachments/assets/34824fbc-43b8-46eb-a38c-c44abb73ca84) | ![Screenshot 2024-12-05 at 5 12 08 PM](https://github.com/user-attachments/assets/c3d1a786-af97-44f0-83ef-07b985e6be0c) |
| Quantity 1 / No allow backorders / Stock management | ![Screenshot 2024-12-05 at 5 18 01 PM](https://github.com/user-attachments/assets/ededd8e5-fe38-4d76-8263-189df78f52ff) | ![Screenshot 2024-12-05 at 5 17 43 PM](https://github.com/user-attachments/assets/9194fd41-57f4-406d-ad65-f8747587599b) |
| Quantity 1 / Allow backorders (notify) / Stock management | ![Screenshot 2024-12-05 at 5 20 59 PM](https://github.com/user-attachments/assets/dc3d4842-1216-4685-b41c-90c709c0017c) | ![Screenshot 2024-12-05 at 5 20 46 PM](https://github.com/user-attachments/assets/2abda81c-ec0e-442b-9c48-3ead87940b7f) | 
| Quantity 1 / Allow backorders (no notify) / Stock management  | ![Screenshot 2024-12-05 at 5 23 17 PM](https://github.com/user-attachments/assets/8e5db468-2300-40ca-951a-0a9e7bb914c5) | ![Screenshot 2024-12-05 at 5 23 02 PM](https://github.com/user-attachments/assets/167b5575-1d7c-4b1c-a410-ddeba16e1901) | 

Closes #53460.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a simple product and keep the product editor open. 

![Screenshot 2024-12-10 at 12 41 44 PM](https://github.com/user-attachments/assets/937a44d4-a457-48c4-a7f2-5be9ca20de0f)


2. In another tab, go to Pages > Add New Page and add a `Product collection` block.
3. Choose `Hand-Picked Products` and select the product you just created.
4. Now add a `Stock indicator` block inside the product you picked.

![Screenshot 2024-12-10 at 10 40 10 AM](https://github.com/user-attachments/assets/a29625a9-2478-47d9-b3b5-b4f7d14b3ac5)

5. Save the page and view it. The stock indicator may be shown (depending on the inventory details described above). E.g.: The indicator won't be visible for simple products `In stock` and without `Stock management`. 
6. Now, view the product page and verify there is no difference between what we show in both stock indicators (inside the Product Collection and the product page).
7. Verify the indicator always matches and consider [this change](https://github.com/woocommerce/woocommerce/pull/53462#discussion_r1872177154).



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
